test/Test/Database/LSMTree/StateMachine.hs

```diff
@@ -1901,8 +1901,10 @@ arbitraryActionWithVars _ label ctx (ModelState st _stats) =
         | let genErrors = pure Nothing -- TODO: generate errors
         ]
      ++ [ (5,  fmap Some $ (Action <$> genErrors <*>) $
-            RangeLookup <$> genRange <*> genTableVar)
+            RangeLookup <$> genRange <*> genNotUnionDescendantTableVar)
+            -- TODO: enable range lookups on tables with unions
         | let genErrors = pure Nothing -- TODO: generate errors
+        , not (null notUnionDescendantTableVars)
```

Comment:
Minor nit: I'd probably reiterate the TODO on the `not (null ...)` case just to be sure we don't forget to remove it

Reply:
yes fair enough, done.